shaders/src/shading_parameters.fs

```diff
@@ -34,6 +34,11 @@ void computeShadingParams() {
 
     shading_position = vertex_worldPosition;
     shading_view = normalize(frameUniforms.cameraPosition - shading_position);
+
+    // we do this so we avoid doing (matrix multiply), but we burn 4 varyings:
+    //    p = clipFromWorldMatrix * shading_position;
+    //    shading_normalizedViewportCoord = p.xy * 0.5 / p.w + 0.5
+    shading_normalizedViewportCoord = vertex_Position.xy * (0.5 / vertex_Position.w) + 0.5;
```

Comment:
Could we store 0.5/w in the varying instead of doing the division here? Or at least 1/w.

Reply:
I don't think you can interpolate 1/w -- I tried to interpolate `.xy/w` (which would be the same) initially and that didn't work.

Reply:
💩